p2p/service/sim.go

```diff
@@ -172,9 +187,18 @@ func (sn *Node) sendMessageImpl(nodeID p2pcrypto.PublicKey, protocol string, pay
 	return errors.New("could not find " + protocol + " handler for node: " + nodeID.String())
 }
 
+func randSleep(delay uint32) {
```

Comment:
it will be easier to reason about what's happening if you would've passed final values for the sleep but if some randomness is still needed you'd generate random number from outside. (that way you can be aware of the random number that we're generated.)